packages/tc-ui/src/primitives/relationship/lib/relationship-picker.jsx

```diff
@@ -128,6 +129,36 @@ class RelationshipPicker extends React.Component {
 		}
 	}
 
+	onChange(event) {
+		const { value, id, checked, dataset, type } = event.currentTarget;
+		const { parentCode } = dataset;
+		const propertyName = id.split('-')[1];
+		this.setState(prevState => {
+			const { selectedRelationships } = prevState;
+			const oldState = selectedRelationships.map(relationship => {
+				if (relationship.code === parentCode) {
+					// for multiple choice values
+					if (type === 'checkbox') {
```

Comment:
would it be possible to keep data manipulations specific to a component in that component? Because, say we need to add a new type of component that returns slightly different kind of data, then we won't need to edit relationship picker - we can just use the component. Biz Ops Admin already defines some custom components - there's a chance they won't work if relationship-picker doesn't understand the data they send, whereas if the pattern is to always massage the data first before forwarding it, then it shouldn't be a problem.  Something like
```jsx
// in the nested component
({onChange}) => {
const localOnChange = data => onChange(manipulate(data))
 return <input onChange={localOnChange} />
}
```

That's the theory anyway - it might not actually be a good idea - let's chat about it sometime